bug_1292234.patch

```diff
@@ -128,74 +128,80 @@ diff --git a/devtools/server/actors/storage.js b/devtools/server/actors/storage.
 +    }
 +  },
 +
-+  async populateStoreMap(host, storagePrincipal, storeMap) {
-+    const db = await ExtensionStorageIDB.open(storagePrincipal);
-+    this.dbConnectionForHost.set(host, db);
-+    const data = await db.get();
++  async getStoragePrincipal(extension) {
++    if (extension.getSharedData("storageIDBBackend")) {
++      return extension.getSharedData("storageIDBPrincipal");
++    }
++    // This extension either isn't using the IDB backend (false) or
++    // it hasn't migrated to the IDB backend yet (null)
 +
-+    for (const [key, value] of Object.entries(data)) {
-+      storeMap.set(key, value);
++    // Loop for web extension context related to the current window
++    let currentContext;
++    for (const view of extension.views) {
++      currentContext = view;
```

Comment:
I was previously getting context from `view.childManager.context` -- how is this different from using the `view` directly?

Reply:
There isn't any actual difference, it's just that `view.childManager.context` is actually a reference to `view`:

- The `view` is an instance of an "extension context" class (e.g. `ExtensionPageContextChild`)
- The extension contexts have a `childManager` property that is lazily created: https://searchfox.org/mozilla-central/rev/2c912888e3b7ae4baf161d98d7a01434f31830aa/toolkit/components/extensions/ExtensionPageChild.jsm#273
- In the childManager's constructor, its `context` property is set to the reference of its related "extension context": https://searchfox.org/mozilla-central/rev/2c912888e3b7ae4baf161d98d7a01434f31830aa/toolkit/components/extensions/ExtensionChild.jsm#923-925